test/index-test.js

```diff
@@ -110,6 +110,14 @@ tests = [{
     m: 'i'
   , o: [{i: 1, o: 2}, {i: 2, o: 2}]
   , e: [{i: 1}, {i: 2}]
+}, {
+    m: 'foo(bar)'
+  , o: { foo: { biz: 'baz' } }
+  , e: { foo: {} },
```

Comment:
I'm on the fence about this one. On one hand the library's primary purpose is to save bytes. (If it were not, why bother filtering out the unnecessary; just don't use it). So, from that perspective, collapsing empty objects seems like the right thing to do. On the other hand I can imagine the client not wanting to check the existence of a parent object before accessing the child (e.g., `if (foo && foo.bar) return foo.bar.baz;`). What do you think; what motivated this change for you?